grouper/entities/permission_grant.py

```diff
@@ -23,3 +23,15 @@
         ("grant_id", Optional[int]),
     ],
 )
+
+# Represents all grants of a specific permission, as a dictionary of users and service accounts who
+# have been granted that permission to lists of all the arguments to that permission that they have
+# been granted.
+AllGrantsOfPermission = NamedTuple(
+    "AllGrantsOfPermission",
+    [("users", Dict[str, List[str]]), ("service_accounts", Dict[str, List[str]])],
+)
+
+# The same, but for every permission, stored in a dictionary mapping permission names to those
+# grants.
+AllGrants = Dict[str, AllGrantsOfPermission]
```

Comment:
I have a vague feeling that this definition is overkill, probably since it can be described entirely (and cleanly) as a type annotation. By contrast, the definition of `AllGrantsOfPermission` does feel valuable since it a) does provide additional information (the key names) and b) would be very ugly if expanded into a type annotation everywhere.

Reply:
Yup, good call. I've now gotten rid of `AllGrants`.